### PR TITLE
Factory plugin/tests

### PR DESCRIPTION
**Pull request by bgeuken**

### CODE REVIEW

File: src/api/spec/models/obs_factory/distribution_strategy_sle15_spec.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+require 'rails_helper'
+
+RSpec.describe ObsFactory::DistributionStrategySLE15 do
+  let(:stategy) { ObsFactory::DistributionStrategySLE15.new(project: create(:project)) }
+
+  describe '#sp_version' do
+    it 'returns the correct service pack version' do
+      expect(stategy.sp_version('SUSE:SLE-15:GA')).to eq(nil)
+      expect(stategy.sp_version('SUSE:SLE-15-SP4:GA')).to eq('4')
```
[the remark this answers is not in the document]
Comment (bgeuken) [reply src/api/spec/models/obs_factory/distribution_strategy_sle15_spec.rb]: It was 'SUSE:SLE-15-4:GA' before. Forgot to change the other half of the test

**Comment by bgeuken** [comment]
Fixed. And also solved the rubocop offenses

File: src/api/spec/models/obs_factory/distribution_strategy_sle15_spec.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+require 'rails_helper'
+
+RSpec.describe ObsFactory::DistributionStrategySLE15 do
+  let(:stategy) { ObsFactory::DistributionStrategySLE15.new(project: create(:project)) }
+
+  describe '#sp_version' do
+    it 'returns the correct service pack version' do
+      expect(stategy.sp_version('SUSE:SLE-15:GA')).to eq(nil)
```
Comment (Ana06) [inline src/api/spec/models/obs_factory/distribution_strategy_sle15_spec.rb]: what about using `to be_nil`? :smile:

File: src/api/spec/models/obs_factory/distribution_strategy_factory_spec.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
+require 'rails_helper'
+
+RSpec.describe ObsFactory::DistributionStrategyFactory do
+  let(:project) { create(:project) }
+
+  describe '::new' do
+    context 'called with no parameters' do
+      subject { ObsFactory::DistributionStrategyFactory.new }
+
+      it { is_expected.to be_kind_of(ObsFactory::DistributionStrategyFactory) }
```
Comment (hennevogel) [inline src/api/spec/models/obs_factory/distribution_strategy_factory_spec.rb]: What are you testing with this? `Class.new`?

File: src/api/spec/models/obs_factory/distribution_strategy_factory_spec.rb [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    end
+
+    context 'called with project parameter' do
+      subject { ObsFactory::DistributionStrategyFactory.new(project: project) }
+
+      it { is_expected.to be_kind_of(ObsFactory::DistributionStrategyFactory) }
+      it { is_expected.to have_attributes(project: project) }
```
Comment (hennevogel) [inline src/api/spec/models/obs_factory/distribution_strategy_factory_spec.rb]: Same here, these are stdlib things...

File: src/api/spec/models/obs_factory/distribution_strategy_factory_spec.rb (hunk 3 above)
Comment (bgeuken) [reply src/api/spec/models/obs_factory/distribution_strategy_factory_spec.rb]: Just have a look some lines [above](https://github.com/openSUSE/open-build-service/pull/5191/files#diff-ad0180b8e7006a09dab526848e938a24R6) :wink:

File: src/api/spec/models/obs_factory/distribution_strategy_factory_spec.rb (hunk 4 above)
Comment (bgeuken) [reply src/api/spec/models/obs_factory/distribution_strategy_factory_spec.rb]: What you mean is not Ruby stdlib, but Rails's [ActiveModel::Model's new](http://api.rubyonrails.org/classes/ActiveModel/Model.html#method-c-new). For Ruby classes you need to define an initializer to add additional behaviour, like assigning a project if you want that.

Thinking about it we could also include ActiveModel::Model to get this behaviour. :thinking:

**Pull request merged**